### PR TITLE
Updated the README

### PR DESCRIPTION
**Pull request by iamthemuffinman**
I added the pat homepage, a code example, a Go Report Card badge, and some notes at the bottom. Much better now. :+1:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,88 @@
-pat
-===
-[![Build Status](https://travis-ci.org/gorilla/pat.png?branch=master)](https://travis-ci.org/gorilla/pat)
+# pat [![Build Status](https://travis-ci.org/gorilla/pat.png?branch=master)](https://travis-ci.org/gorilla/pat) [![Go Report Card](https://goreportcard.com/badge/github.com/gorilla/pat)](https://goreportcard.com/report/github.com/gorilla/pat)
```
Comment (elithrar) [inline README.md]: Copy the badges used for gorilla/mux - https://github.com/gorilla/mux - SVG icons for GoDoc and TravisCI

File: README.md [hunk 2]
```diff
@@ -1,3 +1,88 @@
...
+
+### How to use?
+
+pat is pretty simple. The documentation lives [here.](http://www.gorillatoolkit.org/pkg/pat)
```
Comment (elithrar) [inline README.md]: Pull the period out of the link text.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,88 @@
-pat
-===
-[![Build Status](https://travis-ci.org/gorilla/pat.png?branch=master)](https://travis-ci.org/gorilla/pat)
+# pat [![Build Status](https://travis-ci.org/gorilla/pat.png?branch=master)](https://travis-ci.org/gorilla/pat) [![Go Report Card](https://goreportcard.com/badge/github.com/gorilla/pat)](https://goreportcard.com/report/github.com/gorilla/pat)
+
+### How to use?
```
Comment (elithrar) [inline README.md]: Follow the headings in https://github.com/gorilla/csrf - make this "Example"

File: README.md [hunk 4]
```diff
@@ -1,3 +1,88 @@
-pat
-===
-[![Build Status](https://travis-ci.org/gorilla/pat.png?branch=master)](https://travis-ci.org/gorilla/pat)
+# pat [![Build Status](https://travis-ci.org/gorilla/pat.png?branch=master)](https://travis-ci.org/gorilla/pat) [![Go Report Card](https://goreportcard.com/badge/github.com/gorilla/pat)](https://goreportcard.com/report/github.com/gorilla/pat)
+
```
Comment (elithrar) [inline README.md]: Add an Install section like https://github.com/gorilla/csrf#install

File: README.md [hunk 5]
```diff
@@ -1,3 +1,88 @@
...
+package main
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gorilla/pat"
+)
+
+func homeHandler(wr http.ResponseWriter, req *http.Request) {
```
Comment (elithrar) [inline README.md]: I'd probably make this a little shorter—I don't think there's a need to show as many handlers. 2-3 would suffice to show a parameterized URL, a static URL and a 3 different methods.

**Comment by elithrar** [comment]
Thanks for submitting this. Made some comments—where possible try to align with https://github.com/gorilla/csrf/tree/master/README.md

**Comment by iamthemuffinman** [comment]
@elithrar Let me know if anything should be changed.

**Comment by iamthemuffinman** [comment]
@elithrar Did you have a chance to look this over?

**Comment by elithrar** [comment]
@iamthemuffinman Sorry, been super busy! Can you squash this? The new GitHub feature doesn't seem to be appearing for this PR (which is probably because it predates the feature). I'll merge as soon as you do that.

**Comment by iamthemuffinman** [comment]
@elithrar Squashed. :)

**Comment by elithrar** [comment]
Merging (tests are fine; just the old .travis.yml causing issues)

**Comment by elithrar** [comment]
Thanks @iamthemuffinman!

**Comment by iamthemuffinman** [comment]
You're welcome @elithrar!

**Pull request merged**